File: Assets/Plugins/Android/NativeBridge.java
package com.pj.nativecore;

import android.util.Log;

public class NativeBridge{

    // private static AndroidBridge instance;

    // public static AndroidBridge Instance(){
    //     if(instance == null){
    //         instance = new AndroidBridge();
    //     }
    //     return instance;
    // }

    private NativeBridgeCallback unityCallback;

    public void initialize(NativeBridgeCallback unityCallback){
        this.unityCallback = unityCallback;
    }

    public void send(String message){
        Log.d("AndroidBridge", "android - java - message : " + message);
    }

    public void send(String message, int nonce){
        message += " ~~~ from java ~~~";
        Log.d("AndroidBridge", "android - java - message : " + message + " and return~~");
        this.unityCallback.onReceive(message, nonce);
    }
}

File: Assets/Scripts/NativeBridge.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using PJ.Core.Util;
using System;


namespace PJ.Native
{
    public class NativeBridge : Singleton<NativeBridge>
    {
        private INativeBridge bridge;

        protected override void Initialize()
        {
#if UNITY_ANDROID 
            bridge = new AndroidBridge();
#elif UNITY_IOS
            bridge = new iOSBridge();
#endif 
        }
    
        public void Send(string message)
        {
            bridge?.Send(message);
        }

        public void Send(string message, Action<string> onReceive)
        {
            bridge?.Send(message, onReceive);
        }
    }
}